### ISSUE TITLE
[BUG] lazy load with async-await fails to generating document

### ISSUE THREAD

**Issue by ksrae** [compodoc/compodoc#1096]
I use async-await on lazy-loading routes like following code:

```
import { Routes } from '@angular/router';

export const routes: Routes = [
  {
    path: 'test',
    loadChildren: async () =>
      (await import('./test/test.module')).TestModule
  },
  ...
```

It causes error as generating compodoc

> [15:55:41] parsing        : /Users/workspace/Project/src/app/app.routes.ts
> [15:55:41] Analysing routes definitions and clean them if necessary
> Unhandled Rejection at: Promise {
>   <rejected> InvalidOperationError: Could not find the node's symbol.
>       at InvalidOperationError.BaseError [as constructor] (/usr/local/lib/node_modules/@compodoc/compodoc/node_modules/ts-simple-ast/dist/errors/BaseError.js:7:28)
>       at new InvalidOperationError (/usr/local/lib/node_modules/@compodoc/compodoc/node_modules/ts-simple-ast/dist/errors/InvalidOperationError.js:8:28)
>       at Object.throwIfNullOrUndefined (/usr/local/lib/node_modules/@compodoc/compodoc/node_modules/ts-simple-ast/dist/errors/helpers.js:96:15)
>       at Identifier.Node.getSymbolOrThrow (/usr/local/lib/node_modules/@compodoc/compodoc/node_modules/ts-simple-ast/dist/compiler/common/Node.js:131:23)
>       at RouterParserUtil.cleanFileDynamics (/usr/local/lib/node_modules/@compodoc/compodoc/dist/index-cli-41ed7d22.js:6023:18)
>       at AngularDependencies.getSourceFileDecorators (/usr/local/lib/node_modules/@compodoc/compodoc/dist/index-cli-41ed7d22.js:8684:46)
>       at /usr/local/lib/node_modules/@compodoc/compodoc/dist/index-cli-41ed7d22.js:8492:31
>       at Array.map (<anonymous>)
>       at AngularDependencies.getDependencies (/usr/local/lib/node_modules/@compodoc/compodoc/dist/index-cli-41ed7d22.js:8481:21)
>       at CliApplication.Application.getDependenciesData (/usr/local/lib/node_modules/@compodoc/compodoc/dist/index-cli-41ed7d22.js:10109:40)
> } reason: InvalidOperationError: Could not find the node's symbol.
>     at InvalidOperationError.BaseError [as constructor] (/usr/local/lib/node_modules/@compodoc/compodoc/node_modules/ts-simple-ast/dist/errors/BaseError.js:7:28)
>     at new InvalidOperationError (/usr/local/lib/node_modules/@compodoc/compodoc/node_modules/ts-simple-ast/dist/errors/InvalidOperationError.js:8:28)
>     at Object.throwIfNullOrUndefined (/usr/local/lib/node_modules/@compodoc/compodoc/node_modules/ts-simple-ast/dist/errors/helpers.js:96:15)
>     at Identifier.Node.getSymbolOrThrow (/usr/local/lib/node_modules/@compodoc/compodoc/node_modules/ts-simple-ast/dist/compiler/common/Node.js:131:23)
>     at RouterParserUtil.cleanFileDynamics (/usr/local/lib/node_modules/@compodoc/compodoc/dist/index-cli-41ed7d22.js:6023:18)
>     at AngularDependencies.getSourceFileDecorators (/usr/local/lib/node_modules/@compodoc/compodoc/dist/index-cli-41ed7d22.js:8684:46)
>     at /usr/local/lib/node_modules/@compodoc/compodoc/dist/index-cli-41ed7d22.js:8492:31
>     at Array.map (<anonymous>)
>     at AngularDependencies.getDependencies (/usr/local/lib/node_modules/@compodoc/compodoc/dist/index-cli-41ed7d22.js:8481:21)
>     at CliApplication.Application.getDependenciesData (/usr/local/lib/node_modules/@compodoc/compodoc/dist/index-cli-41ed7d22.js:10109:40)
> [15:55:43] Sorry, but there was a problem during parsing or generation of the documentation. Please fill an issue on github. (https://github.com/compodoc/compodoc/issues/new)

but it works using then() instead of async-await,

```
import { Routes } from '@angular/router';

export const routes: Routes = [
  {
    path: 'test',
    loadChildren:() =>
      import('./test/test.module').then(m=> m.TestModule)
  },
  ...
  
```

can anyone fix to allow using async-await on routes.

<env>
Angular:12.0.1
Node: 14.0

**Comment by vogloblinsky** [comment]
Partial close, this need a complete rewrite of routing parsing engine.

**Comment by vogloblinsky** [comment]
Fixed, full support coming in 1.1.15
